### DISCUSSION TITLE
How to copy everything from one supabase project to another? [Reliable & Easy]

### DISCUSSION THREAD

**Discussion by NikhilCodes** [supabase/supabase#7324] [Questions] [answered]
I have this project in supabase, want to copy everything i.e., RPC functions, tables, users, webhooks etc to another project.
How to do that?

**Comment by silentworks** [comment, marked as answer]
You can do this using the migrate between projects guide https://supabase.com/docs/guides/database#migrating-between-projects

**Reply by benjamindonze** [reply]
Hello the link is no more accurate where can I find this documentation nowadays`?

**Reply by spassos98** [reply]
There is a new link about migrating the databases from one project to another one https://supabase.com/docs/guides/platform/migrating-and-upgrading-projects

**Reply by bowis** [reply]
> There is a new link about migrating the databases from one project to another one https://supabase.com/docs/guides/platform/migrating-and-upgrading-projects

I dont see any mention regarding migration there

**Reply by silentworks** [reply]
@Bowis clearly the URL has changed since it was posted in August 2023, but if you look in the left navigation you would see migrate within supabase. Also please don't be adding thumbs down emoji for something that was correct months ago.

**Reply by joobisb** [reply]
current url

https://supabase.com/docs/guides/platform/migrating-within-supabase

**Comment by jt6677** [comment]
Why cannot we have a left click "duplicate"?

**Reply by DwareLab** [reply]
someone requested for this in 2022 and still nothing

**Reply by pivot-marc** [reply]
Still nothing official in 2026, so I ended up building exactly this: [https://www.supaclone.io/](https://www.supaclone.io/). One click, and it clones a Supabase project into another one (schema, RPC functions, roles, and optionally the data).

Full disclosure: I'm the maker. Built it because I got tired of the manual `pg_dump` dance every time. Happy to answer questions if anyone wants to try it.

**Reply by DenisLC** [reply]
is Supaclone capable to clone a Supabase hosted project to a self-hosted version?

**Comment by hitmoose007** [comment]
The link is not accurate anymore

**Comment by timrodz** [comment]
Instead of running `psql` on step https://supabase.com/docs/guides/platform/migrating-and-upgrading-projects#restore-to-your-new-project, I basically did the following:

Supabase > SQL Editor > New Query

1. Dump `roles.sql` into that query, run everything. If successful, delete contents and proceed to step 2:
2. Dump `schema.sql` into that query, run everything. If successful, delete contents and proceed to step 3:
3. Dump `data.sql` into that query, run everything.

Your DB should now be duplicated. Took about 1 min to do all this.

**Extra steps**:

1. Link your new project ID: `supabase link --project-ref <id-here>`
2. `supabase stop && supabase start` to restart it, just in case
3. Generate types to verify everything's looking fresh: `supabase gen types typescript --project-id "<id-here>" --schema public > src/dir/to/types.ts`

CC @hitmoose007

**Comment by tekmen0** [comment]
I have 2 versions, prod - dev. I want to copy schemas etc. from dev to prod but not data.

**Reply by Aniketkhote** [reply]
Same thing i just want to migrate schema and the rpc

**Reply by hamzaakramdropslab** [reply]
@tekmen0 did you find the solution

**Reply by tekmen0** [reply]
no, manually copying

**Reply by ahmersaud** [reply]
if you still havent found any solutions yet,
here is what  i found[  https://medium.com/@christoph.geiser/how-to-duplicate-supabase-database-into-another-project-eb86dbfbeb06]( https://medium.com/@christoph.geiser/how-to-duplicate-supabase-database-into-another-project-eb86dbfbeb06)
looks like an easy solution instead of manually doing it.

**Reply by maricabertarini** [reply]
how about the configs like email templates, dns settings, authentication setup, etc?
Is there a way to copy all of that without the risk of forgetting to setup some of it?
🙏 👋

**Reply by drewbaker** [reply]
@maricabertarini that is exactly what I need, to clone prod to dev and back again.
